### DISCUSSION TITLE
Unable to search for nested groups

### DISCUSSION THREAD

**Discussion by minderjan** [keycloak/keycloak#12064] [Archive]
Hello, 

the old admin UI provided a capability to search for subgroups. This was a really helpful feature, if there exists many nested child groups. As I've seen in keycloak 18, the new admin UI performs only a local search within the current displayed group list. But this makes it impossible to search for nested groups. 

Probably this would be solved, if there's a possibility to display the whole group tree (mentioned in [12030](https://github.com/keycloak/keycloak/discussions/12030)).

__Example__

_We have one parent and one child group_
![child-group-01](https://user-images.githubusercontent.com/1324909/168981226-cacd7b51-6333-4d50-badc-ffb7083c0915.png)

_I want to search for `child-group-01` on the main group menu_
![search-child-group-01](https://user-images.githubusercontent.com/1324909/168981253-89c8e948-18b7-4bed-a3d8-e13bd9a588e2.png)

__Expected__ The `child-group-01` will be found.

-----

__Old UI__

The current UI performs an API call which returns the whole tree of groups, which matches the search term. Within this tree, all matching groups will be highlighted locally.

![old-ui-search](https://user-images.githubusercontent.com/1324909/168983614-7c9c482b-3074-489f-88fa-5c675e15ac03.png)

**Comment by ssilvert** [comment]
Hi @minderjan .

On the Action menu, there is a "Search group" feature that will search all subgroups.  Does that solve your issue?

**Reply by minderjan** [reply]
Hi @ssilvert, 

Unfortunately I didn't notice the second (overall) search feature. This solves my issue. Thanks for clarifying. 

I've tested the overall group search on my `keycloak 18` instance.

1. Searching for a subgroup works perfectly. The expected group will be found.

![search-result](https://user-images.githubusercontent.com/1324909/169068302-13222bd2-6334-4612-92ad-e36f424444aa.png)

2. If I want to open this group by clicking the group name, a `404` error occurs. (tested in firefox 100 and chrome 101)
![open-group](https://user-images.githubusercontent.com/1324909/169068410-ccddd77f-8ba5-4c4e-a44f-be86c56d812f.png)

I don't know if this behavior is already addressed, or it's related to my setup. For some reason the generated URL behind the group name, contains an unnecessary `/search` part.

```html
<a href="#/realm-name/groups/search/1a8cdbb9-3401-4dcc-8c47-8d32520d5312/faac9ac5-29f6-40ef-84f5-488095ae9127">child-group-01</a>
```

instead of:

```html
<a href="#/realm-name/groups/1a8cdbb9-3401-4dcc-8c47-8d32520d5312/faac9ac5-29f6-40ef-84f5-488095ae9127">child-group-01</a>
```

**Comment by ssilvert** [comment]
@minderjan Yes, that's broken now.  Thanks for the feedback!  We will fix it asap.
